tests/Unit/Domain/Creators/Test_BinaryCompactObject.cpp

```diff
@@ -0,0 +1,360 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <cstddef>
+#include <memory>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"  // IWYU pragma: keep
+#include "Domain/Block.hpp"                       // IWYU pragma: keep
+#include "Domain/BlockNeighbor.hpp"               // IWYU pragma: keep
+#include "Domain/Creators/BinaryCompactObject.hpp"
+#include "Domain/Creators/DomainCreator.hpp"
+#include "Domain/Domain.hpp"
+#include "tests/Unit/Domain/DomainTestHelpers.hpp"
+#include "tests/Unit/TestCreation.hpp"
+
+namespace {
+void test_binary_compact_object_construction(
+    const domain::creators::BinaryCompactObject&
+        binary_compact_object) {
+  const auto domain = binary_compact_object.create_domain();
+  test_initial_domain(domain,
+                      binary_compact_object.initial_refinement_levels());
+  test_physical_separation(binary_compact_object.create_domain().blocks());
+}
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Domain.Creators.BinaryCompactObject.Connectivity",
```

Comment:
[optional] Since you opened this PR, it has become the preference to reduce the number of Catch test cases that appear in an effort to make testing proceed faster on certain systems. So, eventually we'll probably want to combine these several `SPECTRE_TEST_CASE`s into a single one which calls functions defined in the anonymous namespace for the individual tests for the new domain.
If you'd prefer to leave that for future clean-up work, though, I have no objections.